File: Java/Board.java
package bot_Ghost_in_the_Cell;

import java.util.ArrayList;
import java.util.List;

public class Board {
	public Factory[] factories;
	public List<Troop> troops = new ArrayList<Troop>();
	public List<Bomb> bombs = new ArrayList<Bomb>();
	public int turn = -1;
	public int bombsLeft = 2;
	
	public Board(int factoryCount) {
		this.factories = new Factory[factoryCount];		
		for (int i = 0; i < factoryCount; i++) {
			this.factories[i] = new Factory(i);
		}
	}
	
	public void addTroop(Troop troop) {
		this.troops.add(troop);
		troop.destination.incomingTroops.add(troop);
	}
	
	public void addBomb(Bomb bomb) {
		this.bombs.add(bomb);
		bomb.source.incomingBombs.add(bomb);
		if (bomb.destination != null) {
			bomb.destination.incomingBombs.add(bomb);
		}
	}
	
	public void initRound() {
		this.troops.clear();
		this.bombs.clear();
		turn++;
		
		for (Factory factory : this.factories) {
			factory.incomingTroops.clear();
			factory.incomingBombs.clear();
		}
	}
}

File: Javascript/Board.js
const Factory = require('./Factory');
const Troop = require('./Troop');
const Bomb = require('./Bomb');

class Board {
    constructor(factoryCount) {
        this.factories = [];
        this.troops = [];
        this.bombs = [];

        for (let i = 0; i < factoryCount; i++) {
            this.factories[i] = new Factory(i);
        }
        
        this.turn = -1;
        this.bombsAvailable = 2;
    }

    addTroop(owner, sourceID, destinationID, production, remainingTurns) {
        const troop = new Troop(owner, this.factories[sourceID], this.factories[destinationID], production, remainingTurns);
        this.troops.push(troop);
        this.factories[destinationID].incomingTroops.push(troop);
    }

    addBomb(owner, sourceID, destinationID, remainingTurns) {
        const bomb = new Bomb(owner, this.factories[sourceID], destinationID !== -1 ? this.factories[destinationID] : null, remainingTurns);
        this.bombs.push(bomb);
        if (destinationID !== -1) {
            this.factories[destinationID].incomingBombs.push(bomb);
        }
    }

    initRound() {
        this.troops = [];
        this.bombs = [];
        this.factories.forEach(factory => {
            factory.incomingTroops = [];
            factory.incomingBombs = [];
        });

        this.turn++;
    }
}

module.exports = Board;